target_s3_jsonl/__init__.py

```diff
@@ -82,75 +99,66 @@ def persist_lines(messages, config):
     validators = {}
 
     naming_convention_default = '{stream}-{timestamp}.jsonl'
-    naming_convention = config.get('naming_convention') if config.get('naming_convention') is not None else naming_convention_default
+    naming_convention = config.get('naming_convention') or naming_convention_default
     compression = None
 
     if f"{config.get('compression')}".lower() == 'gzip':
         compression = 'gzip'
         naming_convention_default = f"{naming_convention_default}.gz"
         naming_convention = f"{naming_convention}.gz"
 
+    elif f"{config.get('compression')}".lower() == 'lzma':
+        compression = 'lzma'
+        naming_convention_default = f"{naming_convention_default}.xz"
+        naming_convention = f"{naming_convention}.xz"
+
     elif f"{config.get('compression')}".lower() not in {'', 'none'}:
         raise NotImplementedError(
             "Compression type '{}' is not supported. "
-            "Expected: 'none' or 'gzip'"
-            .format(f"{config.get('compression')}".lower())
-        )
+            "Expected: 'none', 'gzip', or 'lzma'"
+            .format(f"{config.get('compression')}".lower()))
 
-    # Use the system specific temp directory if no custom temp_dir provided
+    # NOTE: Use the system specific temp directory if no custom temp_dir provided
     temp_dir = Path(config.get('temp_dir', tempfile.gettempdir())).expanduser()
 
-    # Create temp_dir if not exists
-    if temp_dir:
-        temp_dir.mkdir(parents=True, exist_ok=True)
+    # NOTE: Create temp_dir if not exists
+    temp_dir.mkdir(parents=True, exist_ok=True)
 
-    filenames = []
+    file_data = {}
     now = datetime.now().strftime('%Y%m%dT%H%M%S')
 
     for message in messages:
         try:
             o = json.loads(message)
```

Comment:
I remember now. It all started because `singer-python` couldn't handles the messages with `singer.parse_message`. After further investigation I figured that `singer.parse_message` is broken since `singer-python==5.6.1` (current being `5.12.1`, so I had to downgrade the version which downgraded `jsonschema` as a dependency causing issues with other thing. I assumed that to be the main reason people moved away from it. Looking at the code `singer.parse_message` uses `json.loads(msg, use_decimal=True)` where [`use_decimal` doesn't seem to exist anymore as an argument](https://docs.python.org/3/library/json.html#json.loads) when I manually tried it.